### ISSUE TITLE
[SF][A11y][ui5-list]: Availability of interactive elements within the row is not getting announced

### ISSUE THREAD

**Issue by Neeeko** [SAP/ui5-webcomponents#7729]
### Bug Description

Using a screen reader such as JAWS, when we focus on a full row which contains a button, the screen reader properly announces the content of the row, but not the interactive element. Ideally, it should announce a button as such so that an accessible user can understand that there's an interactive element within the row.
<img width="1728" alt="Screenshot 2023-10-18 at 6 15 30 PM" src="https://github.com/SAP/ui5-webcomponents/assets/7316793/7bb5b2af-b13d-4e29-87b9-24ae191bda59">

This feature works properly in SAPUI5: https://sapui5.hana.ondemand.com/#/entity/sap.m.CustomListItem/sample/sap.m.sample.CustomListItem

But not in this code sand box: https://codesandbox.io/s/v63xqt?file=/index.html

Another item missing from the SAPUI5 library is the fact that when the focus is on the row itself, pressing F2 puts the focus on the first focusable element within the row, which is not the case in ui5-webcomponents.

You may refer to this video from our a11y experts:
https://github.com/SAP/ui5-webcomponents/assets/7316793/b7f8af19-dec9-4c3d-9cd3-5db9740832e9

### Affected Component

ui5-list

### Isolated Example

https://codesandbox.io/s/v63xqt?file=/index.html

### Steps to Reproduce

1. Go to https://codesandbox.io/s/v63xqt?file=/index.html
2. With JAWS (or NVDA), focus on the row
3. Note that only the button label is announced but not the fact that it's an interactive element
4. Press F2
5. Note that the focus doesn't go to the first interactive element inside the row

### Priority

Medium

### UI5 Web Components Version

latest

### Browser

Chrome, Edge, Firefox, Safari

### Operating System

macOS

### Organization

SuccessFactors

### Declaration

- [X] I’m not disclosing any internal or sensitive information.

**Comment by yanaminkova** [comment]
Hi @Neeeko,

Could you please split the described items into two separate issues, so we could process them more precisely?

Thanks and Kind regards,
Yana

**Comment by Neeeko** [comment]
Hi @yanaminkova, I raised the following defect for the F2 issue: https://github.com/SAP/ui5-webcomponents/issues/7736

**Comment by niyap** [comment]
Hello @SAP/ui5-webcomponents-topic-p,

When we have CustomListItem with interactive element within it, for example ui5-button, when the list item is on focus, just the button's text is announced. There is no information that it is a button. : https://sap.github.io/ui5-webcomponents/nightly/playground/main/pages/List/
In UI5, in the same scenario both "button" and its text is announced.

Could you please look over?

Thank you in advance!

Kind Regards,
Niya

**Comment by yanaminkova** [comment]
Hello @Neeeko,

I wanted to inform you that we are currently discussing how to address this feature request. 
We will update you once a final decision has been made and we begin implementing it.

Kind regards,
Yana

**Comment by Neeeko** [comment]
hi @dobrinyonkov , any plan on when this will be addressed?

**Comment by judith9209** [comment]
Brought up this bug today 2025-10-09 in UI5 monthly planning meeting to ask for priority.

**Comment by simlin** [comment]
FIORITECHE1-9519

**Comment by dobrinyonkov** [comment]
This is currently being worked on with BGSOFUIPIRIN-6946
